modules/integration_azure-api-management-service/detectors-api_management_service.tf

```diff
@@ -0,0 +1,113 @@
+resource "signalfx_detector" "heartbeat" {
+  name = format("%s %s", local.detector_name_prefix, "Azure Api Management Service heartbeat")
+
+  program_text = <<-EOF
+        from signalfx.detectors.not_reporting import not_reporting
+        base_filter = filter('resource_type', 'Microsoft.ApiManagement/service') and filter('primary_aggregation_type', 'true')
+        signal = data('Requests', filter=base_filter and ${module.filter-tags.filter_custom})${var.heartbeat_aggregation_function}.publish('signal')
```

Comment:
I don't think request is a good metric for the healthcheck since an unused Api Management will trigger an alert.
Maybe `NetworkConnectivity` ?